contracts/modules/cover/Cover.sol

```diff
@@ -534,54 +531,6 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     return coverId;
   }
 
-  function createStakingPool(
-    bool isPrivatePool,
-    uint initialPoolFee,
-    uint maxPoolFee,
-    ProductInitializationParams[] memory productInitParams,
-    string calldata ipfsDescriptionHash
-  ) external whenNotPaused onlyMember returns (uint /*poolId*/, address /*stakingPoolAddress*/) {
-
-    uint numProducts = productInitParams.length;
-
-    // override with initial price and check if pool is allowed
-    for (uint i = 0; i < numProducts; i++) {
-
-      if (productInitParams[i].targetPrice < GLOBAL_MIN_PRICE_RATIO) {
-        revert TargetPriceBelowGlobalMinPriceRatio();
-      }
-
-      uint productId = productInitParams[i].productId;
-
-      // if there is a list of allowed pools for this product - this pool didn't exist yet so it's not in it
-      if (allowedPools[productId].length > 0) {
-        revert PoolNotAllowedForThisProduct(productId);
-      }
-
-      if (productId >= _products.length || _products[productId].isDeprecated) {
-        revert ProductDoesntExistOrIsDeprecated();
-      }
-
-      productInitParams[i].initialPrice = _products[productId].initialPriceRatio;
-    }
-
-    (uint poolId, address stakingPoolAddress) = stakingPoolFactory.create(address(this));
-
-    IStakingPool(stakingPoolAddress).initialize(
-      isPrivatePool,
-      initialPoolFee,
-      maxPoolFee,
-      poolId,
-      ipfsDescriptionHash
-    );
-
-    tokenController().assignStakingPoolManager(poolId, msg.sender);
-
-    stakingProducts().setInitialProducts(poolId, productInitParams);
```

Comment:
💡 idea: 
1. we move the `createStakingPool` function to `CoverProducts.sol`
2. we remove the functionality to add products in the same tx (i.e. `stakingProducts().setInitialProducts(poolId, productInitParams);`) - the UX is similar - 2 txs instead of 1; we remove the dependency between SP and CP and we get rid of duplicate code in SP

Reply:
Removing dependency of SP on to CP is still there because it checks for requirePoolIsAllowed in setProducts and productsCount in recalculateEffectiveWeightsForAllProducts

Do you still see value in separating it into 2 transactions  and moving it to CoverProducts?

Reply:
There's one more issue with moving the `createStakingPool` function - currently the `Cover` is the operator that is allowed to call the `StakingPoolFactory` contract. The operator can be changed, however I don't think the overhead is worth it - `Cover` is also a source for some calls in the `StakingPool` contracts, size constraints are gone given the products being moved, do you also think it would be easier to just leave it here?

Reply:
I'd prefer to move it out  to StakingProducts just to ensure we have plenty space in Cover.sol going forward and we don't have to think about its size again

Changing the tests to adapt to this involves lot of minute changes, so if we decide to move it out again we'd have to apply those changes again